src/test/java/teammates/test/cases/util/ActivityLogGeneratorTest.java

```diff
@@ -0,0 +1,275 @@
+        
+        // Masquerade: userType and acc don't have the same google id
+        generatedMessage = logCenter.generatePageActionLogMessage(url, mockParamMap, userType, acc, null, "View comments");
+        AssertHelper.assertLogMessageEquals(logMessage, generatedMessage);
+        
+    }
+    
+    @Test
+    public void testGenerateActivityLogFromAppLogLine() {
+        ______TS("Success: Generate activityLog from appLogLine (with TimeTaken)");
+        String logMessageWithoutTimeTaken = "TEAMMATESLOG|||instructorHome|||Pageload|||true|||Instructor"
+                                            + "|||UserName|||UserId|||UserEmail|||Message|||URL|||UserId20151019143729608";
+        AppLogLine appLog = new AppLogLine();
+        appLog.setLogMessage(logMessageWithoutTimeTaken + Const.ActivityLog.FIELD_SEPARATOR + "20");
+        ActivityLogEntry entry = logCenter.generateActivityLogFromAppLogLine(appLog);
+        assertEquals(logMessageWithoutTimeTaken, entry.generateLogMessage());
+        assertEquals(20, entry.getActionTimeTaken());
+        
+        ______TS("Success: Generate activityLog from appLogLine (without TimeTaken)");
+        appLog.setLogMessage(logMessageWithoutTimeTaken);
+        entry = logCenter.generateActivityLogFromAppLogLine(appLog);
+        assertEquals(logMessageWithoutTimeTaken, entry.generateLogMessage());
+        assertEquals(0, entry.getActionTimeTaken());
+        
+        ______TS("Success with severe log: timeTaken not in correct format");
+        appLog.setLogMessage(logMessageWithoutTimeTaken + Const.ActivityLog.FIELD_SEPARATOR + "random");
+        entry = logCenter.generateActivityLogFromAppLogLine(appLog);
+        assertEquals(logMessageWithoutTimeTaken, entry.generateLogMessage());
+        assertEquals(0, entry.getActionTimeTaken());
+        
+        ______TS("Fail: log message not in correct format");
+        appLog.setLogMessage("TEAMMATESLOG||RANDOM");
+        entry = logCenter.generateActivityLogFromAppLogLine(appLog);
+        assertTrue(entry.generateLogMessage().contains(Const.ActivityLog.MESSAGE_ERROR_LOG_MESSAGE_FORMAT));
+    }
+    
+    private Map<String, String[]> generateMockParamsWithRegKey() {
+        Map<String, String[]> params = new HashMap<String, String[]>();
+        params.put(Const.ParamsNames.COURSE_ID, new String[] { "CS2103" });
+        params.put(Const.ParamsNames.STUDENT_EMAIL, new String[] { "student@email.com" });
+        params.put(Const.ParamsNames.REGKEY, new String[] { "KeyABC" });
+        return params;
+    }
+
+    private EmailWrapper generateMockEmailWrapperFromRequest(Throwable t) {
+        EmailWrapper errorReport =
+                new EmailGenerator().generateSystemErrorEmail("GET", "MAC OS", "/page/somePage",
+                                                              "http://example/", "{}", null, t);
+        return errorReport;
```

Comment:
Two things here:
- This is not really a mock, isn't it? It's calling the actual email-generating function.
- Simply `return new EmailGenerator()....` will make it shorter.